src/mono/mono/mini/mini.c

```diff
@@ -3491,6 +3491,9 @@ mini_method_compile (MonoMethod *method, guint32 opts, JitFlags flags, int parts
 	 */
 	mono_compile_create_vars (cfg);
 
+	if (cfg->exception_type == MONO_EXCEPTION_TYPE_LOAD)
+		cfg->exception_type = MONO_EXCEPTION_NONE;
+
```

Comment:
This feels fragile. What happens if we don't clear out the exception?  Is the intention to only clear the error if it has been "handled" in some way?  If so, why don't we clear it when we handle it?

Reply:
I agree. At this point I wanted to see CI results, so there are scattered instances of clearing the exception just to get it working. I will reduce their number and move them to more logical places.